pkg/cmd/roachtest/tests/cdc.go

```diff
@@ -1731,6 +1731,122 @@ func runMessageTooLarge(ctx context.Context, t test.Test, c cluster.Cluster) {
 	require.Regexp(t, `mvcc=[\d\.]+,\d+`, logStr, "log should include mvcc")
 }
 
+type multiTablePTSBenchmarkParams struct {
+	numRows   int
+	numTables int
+	duration  string
+}
+
+// runCDCMultiTablePTSBenchmark is a benchmark for changefeeds with multiple tables,
+// focusing on the performance of the PTS system. It starts a bank workload on every
+// table it creates and then runs a single changefeed that targets all of these bank tables.
+// Each of those workloads (there will be one per table) will run for the duration specified
+// in the params and have the number of rows specified in the params.
+func runCDCMultiTablePTSBenchmark(
+	ctx context.Context, t test.Test, c cluster.Cluster, params multiTablePTSBenchmarkParams,
+) {
+	ct := newCDCTester(ctx, t, c)
+	defer ct.Close()
+
+	startOpts := option.DefaultStartOpts()
+	startOpts.RoachprodOpts.ExtraArgs = append(startOpts.RoachprodOpts.ExtraArgs,
+		"--vmodule=changefeed=2",
+		"--vmodule=changefeed_processors=2",
+		"--vmodule=protected_timestamps=2",
+	)
+
+	db := ct.DB()
+	if err := setClusterSettingForMultiTablePTSBenchmark(db); err != nil {
+		t.Fatalf("failed to set cluster settings: %v", err)
+	}
+
+	initCmd := fmt.Sprintf("./cockroach workload init bank --rows=%d --num-tables=%d {pgurl%s}",
+		params.numRows, params.numTables, ct.crdbNodes.RandNode())
+	if err := c.RunE(ctx, option.WithNodes(ct.workloadNode), initCmd); err != nil {
+		t.Fatalf("failed to initialize bank tables: %v", err)
+	}
+
+	ct.workloadWg.Add(1)
+	ct.mon.Go(func(ctx context.Context) error {
+		defer ct.workloadWg.Done()
+		workloadCmd := fmt.Sprintf("./cockroach workload run bank --rows=%d --duration=%s --num-tables=%d {pgurl%s}",
+			params.numRows, params.duration, params.numTables, ct.crdbNodes)
+		return c.RunE(ctx, option.WithNodes(ct.workloadNode), workloadCmd)
+	})
+
+	// We generate and run the changefeed, which requires rangefeeds to be enabled.
+	if _, err := db.Exec("SET CLUSTER SETTING kv.rangefeed.enabled = true"); err != nil {
+		t.Fatalf("failed to enable rangefeeds: %v", err)
+	}
+
+	targetNames := make([]string, 0, params.numTables)
+	for i := 0; i < params.numTables; i++ {
```

Comment:
nit: replace with for-range loop
```suggestion
	for i := range params.numTables {
```